### PR TITLE
Add availabilities for each candidate

### LINKED ISSUE #71: Add availabilities for each candidate

### PR DESCRIPTION
**Pull request by LeeEnEn**
- Added a new prefix avail/ to capture the available dates of each candidate.
- Dates are currently as: 1 (Monday) to 7 (Sunday).
- Have yet to update DG and UG

Closes #71

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Availability.java
Comment (lzan98) [inline src/main/java/seedu/address/model/person/Availability.java:14]: Nice use of regex!

File: src/main/java/seedu/address/model/person/Availability.java
Comment (lzan98) [inline src/main/java/seedu/address/model/person/Availability.java:32]: Just a tiny nitpick, maybe could use `isValidDay` instead of `isValidDate` in subsequent PRs, since we are specifying day instead of date - also, I am likely to use `isValidDate` for scheduling 😝

**Review by lzan98 — APPROVED** [review]
LGTM! Left one minor suggestion for the method `isValidDate`, can be updated now or in future PRs.

Shall approve my review first and see if Dom has any more feedback to add before he merges.

File: src/main/java/seedu/address/model/person/Availability.java
Comment (LeeEnEn) [reply src/main/java/seedu/address/model/person/Availability.java:32]: Day sounds more in line with what we want, I will keep that in mind 😄

File: src/main/java/seedu/address/logic/parser/CliSyntax.java
Comment (domlimm) [inline src/main/java/seedu/address/logic/parser/CliSyntax.java:19]: I know we discussed it to be `avail/`.

However, was thinking maybe we could use `free/` instead. Though it's just 1 character shorter but it has a more straightforward meaning than `avail/` when the user looks at that.

To do it in v1.3. Let's discuss En En! 😄 

Regardless, really great job with this functionality! 👍🏻 💯

File: src/main/java/seedu/address/model/person/Availability.java
Comment (domlimm) [reply src/main/java/seedu/address/model/person/Availability.java:14]: I agree @lzan98! Good job @LeeEnEn! Smart pick on using regex rather than iterating through an array.

File: src/main/java/seedu/address/model/person/Availability.java
Comment (domlimm) [reply src/main/java/seedu/address/model/person/Availability.java:32]: +1! For v1.3. Let's KIV!

File: src/test/java/seedu/address/model/person/AvailabilityTest.java
Comment (domlimm) [inline src/test/java/seedu/address/model/person/AvailabilityTest.java:39]: Good job! A variety of absolutely valid test cases! Will really help for the PE. 💯 💯 💯

**Review by domlimm — APPROVED** [review]
Again, good job on this @LeeEnEn!

Some things to KIV for v1.3, not just for this portion but as a whole of the direction that we're heading towards for the demo.

Approved as leon have, merging now! 👍🏻

File: src/main/java/seedu/address/logic/parser/CliSyntax.java
Comment (LeeEnEn) [reply src/main/java/seedu/address/logic/parser/CliSyntax.java:19]: It is possible, although the prefix is usually an abbreviation of the actual word. 🤔 
But up for discussion as well!

File: src/main/java/seedu/address/model/person/Availability.java
Comment (LeeEnEn) [reply src/main/java/seedu/address/model/person/Availability.java:52]: Agreed, it will look much neater than the current presentation. Let's bring it up for discussion.

**Pull request merged**
